hil/ext/switches/juniper.py

```diff
@@ -0,0 +1,404 @@
+# IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied.  See the License for the specific language
+# governing permissions and limitations under the License.
+"""A switch driver for the juniper QFX series.
+
+But in theory can work for any Juniper switch that supports NETCONF.
+Driver uses the python library PyEZ made available by Juniper.
+By default it sets NETCONF session over ssh, i.e. make RPC calls over ssh.
+For this to work, make sure that the Juniper switch is configured
+appropriately to recieve NETCONF connections. Its default port is 830.
+"""
+
+import re
+import logging
+import schema
+import json
+import sys
+import os
+
+from jnpr.junos import Device
+from jnpr.junos.exception import ConnectError
+from jnpr.junos.utils.config import Config
+from jnpr.junos.utils.config import ConfigLoadError
+
+
+from hil.model import db, Switch
+from hil.model import BigIntegerType
+from hil.migrations import paths
+from os.path import dirname, join
+from hil.migrations import paths
+from hil.ext.switches.junos.config_tables.ConfigTables import (
+        InterfaceConfigTable
+        )
+logger = logging.getLogger(__name__)
+paths[__name__] = join(dirname(__file__), 'migrations', 'juniper')
+
+
+class VlanAddError(Exception):
+    pass
+
+
+class ConfigCommitError(Exception):
+    pass
+
+
+class Juniper(Switch):
+    api_name = 'http://schema.massopencloud.org/haas/v0/switches/juniper'
+    dir_name = os.path.dirname(__file__)
+    __mapper_args__ = {
+        'polymorphic_identity': api_name,
+    }
+
+    id = db.Column(
+            BigIntegerType, db.ForeignKey('switch.id'), primary_key=True
+            )
+    hostname = db.Column(db.String, nullable=False)
+    username = db.Column(db.String, nullable=False)
+    password = db.Column(db.String, nullable=False)
+
+    enable_interface = None
+    add_vlan = None
+    remove_vlan = None
+    set_native_vlan = None
+    remove_native_vlan = None
+    remove_default_vlan = None
+    set_trunk_mode = None
+
+    @staticmethod
+    def validate(kwargs):
+        schema.Schema({
+            'username': basestring,
+            'hostname': basestring,
+            'password': basestring,
+        }).validate(kwargs)
+
+    def session(self):
+        return self
+
+    def disconnect(self):
+        pass
+
+    @staticmethod
+    def validate_port_name(port):
+        pass
+
+    def _create_session(self):
+        dev = Device(
+                host=self.hostname, user=self.username, passwd=self.password
+                )
+        dev.bind(cfg=Config)
+        return dev
+
+    def _set_commands(self, port, vlan_id):
+        """ Set of common `set` commands used by other functions."""
+        port_set = "set interfaces {port}".format(port=port)
+        port_del = "delete interfaces {port}".format(port=port)
+
+        self.enable_interface = """
+        {port_del} disable""".format(port_del=port_del)
```

Comment:
Interesting you should say that. I found the using `\n` more distracting. 
This was cleaner in my opinion. Everything between the """ ... """ is statement string. Each new statement starts on a new line.

Reply:
It just looks odd visually; my brain wants to parse the second line as its own statement. Perhaps not worth worrying too much about though since I've asked you to remove these variables anyway.